ortools/linear_solver/xpress_interface.cc

```diff
@@ -1911,4 +1917,19 @@ void XPRS_CC optimizermsg(XPRSprob prob, void* data, const char *sMsg, int nLen,
 	}
 }
 
+void XpressInterface::SetHint() {
+  // Currently the XPRESS API does not handle clearing out previous hints
+  int const length = solver_->solution_hint_.size();
+  auto* const colind = new int[length];
+  auto* const solval = new double[length];
+  for (int i  = 0; i < length ; ++i) {
+    colind[i] = solver_->solution_hint_[i].first->index();
+    solval[i] = solver_->solution_hint_[i].second;
+  }
+  int status = XPRSaddmipsol(mLp, length, solval, colind, "USER_HINT");
+  if (status) {
```

Comment:
Unfortunately, the return status for `XPRSaddmipsol` is undocumented. Do you have any info ?

Reply:
Also, C++17 allows
```suggestion
  if (int status = XPRSaddmipsol(mLp, length, solval, colind, "USER_HINT")) {
```